java/code/src/com/redhat/rhn/frontend/xmlrpc/BaseHandler.java

```diff
@@ -136,6 +141,15 @@ public Object invoke(String methodCalled, List params) throws XmlRpcFault {
             }
         }
 
+        try {
```

Comment:
We should consider if we should continue to have the read-only user logic or if that should/can be replaced by a User group. 
I think we must replace the read-only API user with a new group in the logic that is being implemented in the RBAC, and in that case, the code that is placed a few lines above (L18-L142) should be removed.

Reply:
Yes, the read-only logic can be fully replaced with RBAC. And yes, there'll be many obsolete parts then :)